example/sketch.js

```diff
@@ -13,12 +13,16 @@ function setup() {
   background(255);
   let boundary = new Rectangle(200, 200, 200, 200);
   qtree = new QuadTree(boundary, 4);
+
   for (let i = 0; i < 300; i++) {
     let x = randomGaussian(width / 2, width / 8);
     let y = randomGaussian(height / 2, height / 8);
     let p = new Point(x, y);
     qtree.insert(p);
   }
+
+  mouseX = 200;
+  mouseY = 200;
```

Comment:
Isn't it better to just check for the existence of `mouseX` and `mouseY` in the `draw()` function instead?

Reply:
I think mouseX and Y always exist but are set to 0 by default. So we just need to default them to what we want ^^

Reply:
Moreover, it's better to initialize once than to check every frame, performance wise...

Reply:
With the issue you talk about, `mouseX` and `mouseY` should **always** be checked to know whether they're contained in the canvas.
I'm not sure a simple "if" with some `===`s to check if the coordinates are valid would decrease performances that much, especially if it's to secure something as important as user-input (that we shouldn't trust 😉 )

Reply:
There is no point to check if mouse is in the canvas because if not, it won't return anything already because there won't be any quadtree containing it since they are contained in the canvas ^^

Reply:
I prefer not to mess with the built-in variables in p5 `mouseX` and `mouseY`. I don't mind the weird behavior of adding an extra `0,0` point in this example.